stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -728,6 +728,35 @@ end
 mul!(C::AbstractVector, A::Adjoint{<:Any,<:AbstractTriangular}, B::Transpose{<:Any,<:AbstractVecOrMat}) = throw(MethodError(mul!, (C, A, B)))
 mul!(C::AbstractVector, A::Transpose{<:Any,<:AbstractTriangular}, B::Transpose{<:Any,<:AbstractVecOrMat}) = throw(MethodError(mul!, (C, A, B)))
 
+# preserve triangular structure in in-place multiplication
+for (cty, aty, bty) in ((:UpperTriangular, :UpperTriangular, :UpperTriangular),
+                        (:UpperTriangular, :UpperTriangular, :UnitUpperTriangular),
+                        (:UpperTriangular, :UnitUpperTriangular, :UpperTriangular),
+                        (:UnitUpperTriangular, :UnitUpperTriangular, :UnitUpperTriangular),
+                        (:LowerTriangular, :LowerTriangular, :LowerTriangular),
+                        (:LowerTriangular, :LowerTriangular, :UnitLowerTriangular),
+                        (:LowerTriangular, :UnitLowerTriangular, :LowerTriangular),
+                        (:UnitLowerTriangular, :UnitLowerTriangular, :UnitLowerTriangular))
+    @eval begin
+        function mul!(C::$cty, A::$aty, B::$bty, alpha::Number, beta::Number)
+            m, n = size(B, 1), size(B, 2)
+            if m != size(A, 1)
+                throw(DimensionMismatch("right hand side B needs first dimension of size $(size(A,1)), has size $m"))
+            end
+            if C === A || C === B
+                throw(ArgumentError("output matrix must not be aliased with input matrix"))
+            end
+
+            @views for i in 1:n
+                C[:, i] = alpha*A*B[:, i] + beta*C[:, i]
+            end
+            return C
+        end
+
+        mul!(C::$cty, A::$aty, B::$bty) = mul!(C, A, B, true, false)
```

Comment:
```suggestion
```

This is the generic fallback anyway.

Reply:
Hm, I see why you need that: because you need a more specific method than the one line 700, right?